### PR TITLE
URL of shadow pages are not delivered in the urls array

### PR DESCRIPTION
**Pull request by wachterjohannes**
if you have a shadow page the url for that in the language switcher on the webpage.

__tasks:__

- [x] test coverage
- [ ] gather feedback for my changes

__informations:__

| q                | a
| ---------------- | ---
| Fixed tickets    | none
| BC breaks        | none
| Documentation PR | none

### CODE REVIEW

File: src/Sulu/Component/Content/Mapper/ContentMapper.php [hunk 1]
```diff
@@ -1640,13 +1640,13 @@ private function getLocalizedUrlsForPage(Page $page, NodeInterface $node, $websp
...
         foreach ($webspace->getAllLocalizations() as $localization) {
-
             // prepare translation vars
             $locale = $localization->getLocalization();
+            $shadowLocale = $this->getShadowLocale($node, $localization->getLocalization());
             $translatedProperty = new TranslatedProperty($property, $locale, $this->languageNamespace);
 
             // state property
-            $propertyTranslator = $this->createPropertyTranslator($localization);
+            $propertyTranslator = $this->createPropertyTranslator($shadowLocale);
```
Comment (danrot) [inline src/Sulu/Component/Content/Mapper/ContentMapper.php]: How does this work if it is no shadow page?

Comment (wachterjohannes) [inline src/Sulu/Component/Content/Mapper/ContentMapper.php]: if it is no shadow the requested language is used ... perhaps i should use another name for this variable!

File: src/Sulu/Component/Content/Mapper/ContentMapper.php [hunk 2]
```diff
@@ -1640,13 +1640,13 @@ private function getLocalizedUrlsForPage(Page $page, NodeInterface $node, $websp
         $contentType = $this->contentTypeManager->get($property->getContentTypeName());
 
         foreach ($webspace->getAllLocalizations() as $localization) {
-
             // prepare translation vars
             $locale = $localization->getLocalization();
+            $shadowLocale = $this->getShadowLocale($node, $localization->getLocalization());
```
Comment (wachterjohannes) [inline src/Sulu/Component/Content/Mapper/ContentMapper.php]: @danrot this function returns the shadow locale or the second parameter ... so if the page is a shadow it returns the shadow locale else the requested locale

File: src/Sulu/Component/Content/Mapper/ContentMapper.php (hunk 1 above)
Comment (danrot) [inline src/Sulu/Component/Content/Mapper/ContentMapper.php]: Yeah, that's a bit confusing. Maybe also rename the method name to something like `getActualLocale`? Or do you have a better idea?

Comment (wachterjohannes) [inline src/Sulu/Component/Content/Mapper/ContentMapper.php]: No not really ... perhaps `getResolvedLocale`?

**Comment by wachterjohannes** [comment]
@danrot fixed

File: src/Sulu/Component/Content/Mapper/ContentMapper.php (hunk 1 above)
Comment (wachterjohannes) [inline src/Sulu/Component/Content/Mapper/ContentMapper.php]: fixed

File: src/Sulu/Component/Content/Mapper/ContentMapper.php [hunk 3]
```diff
@@ -1333,7 +1333,7 @@ public function loadByNode(
             false
         );
 
-        $availableLocalization = $this->getShadowLocale($contentNode, $availableLocalization);
+        $availableLocalization = $this->getResolvedLocale($contentNode, $availableLocalization);
         // END: getAvailableLocalization
```
Comment (danrot) [inline src/Sulu/Component/Content/Mapper/ContentMapper.php]: Not related to this PR, but can you remove this comment?

File: src/Sulu/Component/Content/Mapper/ContentMapper.php [hunk 4]
```diff
@@ -1333,7 +1333,7 @@ public function loadByNode(
             false
         );
 
-        $availableLocalization = $this->getShadowLocale($contentNode, $availableLocalization);
+        $availableLocalization = $this->getResolvedLocale($contentNode, $availableLocalization);
```
Comment (danrot) [inline src/Sulu/Component/Content/Mapper/ContentMapper.php]: Should we also call this variable `$resolvedLocalization` then? It's not like it is the only one available..

Comment (wachterjohannes) [inline src/Sulu/Component/Content/Mapper/ContentMapper.php]: this overrides the variable before ... so i cant rename it!

Comment (wachterjohannes) [inline src/Sulu/Component/Content/Mapper/ContentMapper.php]: and thats the same reason for all other appearances! i have searched for it.

**Comment by wachterjohannes** [comment]
@danrot fixed

File: src/Sulu/Component/Content/Mapper/ContentMapper.php (hunk 4 above)
Comment (dantleech) [inline src/Sulu/Component/Content/Mapper/ContentMapper.php]: btw. languages, localization, languageCode == locale.

**Pull request merged**